internal/executor/service/stuck_pod_detector.go

```diff
@@ -63,104 +67,119 @@ func (d *StuckPodDetector) onStuckPodDetected(pod *v1.Pod) (err error, retryable
 	return err, retryable, message
 }
 
-func (d *StuckPodDetector) onStuckPodDeleted(jobId string, record *podRecord) (resolved bool) {
+func (d *StuckPodDetector) onStuckPodDeleted(record *stuckJobRecord) (resolved bool) {
+	// this method is executed after stuck pod was deleted from the cluster
 	if record.retryable {
 		err := d.jobLeaseService.ReturnLease(record.pod)
 		if err != nil {
-			log.Errorf("Failed to return lease for job %s because %s", jobId, err)
+			log.Errorf("Failed to return lease for job %s because %s", record.job.JobId, err)
 			return false
 		}
 
 		leaseReturnedEvent := reporter.CreateJobLeaseReturnedEvent(record.pod, record.message, d.clusterContext.GetClusterId())
 
 		err = d.eventReporter.Report(leaseReturnedEvent)
 		if err != nil {
-			log.Errorf("Failed to report lease returned for job %s because %s", jobId, err)
+			log.Errorf("Failed to report lease returned for job %s because %s", record.job.JobId, err)
 			// We should fall through to true here, as we have already returned the lease and the event is just for reporting
 			// If we fail, we'll try again which could be complicated if the same executor leases is again between retries
 		}
 
 	} else {
 		// Reporting failed even can fail with unfortunate timing of executor restarts, in that case lease will expire and job can be retried
 		// This is preferred over returning Failed event early as user could retry based on failed even but the job could be running
-		event := reporter.CreateJobFailedEvent(record.pod, record.message, map[string]int32{}, d.clusterContext.GetClusterId())
-		err := d.eventReporter.Report(event)
-		if err != nil {
-			return false
+		for _, pod := range record.job.Pods {
+			message := record.message
+			if pod != record.pod {
+				message = "Peer pod stuck."
```

Comment:
what happens if multiple pods in same job all get stuck at the same time?

Reply:
Actually there is always one running job record, effectively first pod to go wrong in the job is selected (https://github.com/G-Research/armada/pull/480/files/0cf26b9ebfe87e9b12b73392b80b8c4c8cbda6f7#diff-c2b7834011d307455bdc4ffc73d04123214ccd7afb6c7a9c3cf74725082c8f62R121).